### ISSUE TITLE
[BREAKING] 0.4.8 hashtable properties are case-sensitive

### ISSUE THREAD

**Issue by f-bader** [cloudbase/powershell-yaml#161]
In the latest version `0.4.8` the resulting hastable is created in a way where the keys are looked-up in a case-sensitive manner. However, PowerShell is case-insensitive in nature and it is best to create hashtables with case-insensitive key look-up.

This can achieved using [IEqualityComparer](https://learn.microsoft.com/en-us/powershell/utility-modules/psscriptanalyzer/rules/useliteralinitializerforhashtable)

If this is intentional it would be great to point it out as a breaking change

**Comment by gabriel-samfira** [comment]
Hi @f-bader 

This is a duplicate of:
  * https://github.com/cloudbase/powershell-yaml/issues/156

The change was intentional (details in the above issue), and the fact that there was no warning on my part is bad, and will change in the future.

TL;DR so you don't have to read the whole issue:
  * The change has been reverted
  * A new version will be published today
  * The change will be added back after the holidays
  * A new release "protocol" will be created with pre-releases before a final release (if possible please help test out future pre-releases if possible).

I will leave this open until the new release is published today.

**Comment by gabriel-samfira** [comment]
Articles like this: https://learn.microsoft.com/en-us/powershell/utility-modules/psscriptanalyzer/rules/useliteralinitializerforhashtable?view=ps-modules

Is why I started having gray hairs before my time. Case insensitivity was a mistake. Microsoft telling people to write code that relies on case insensitivity is mind boggling. Especially considering it can lead to data loss if the data source you're consuming is inherently case sensitive (which YAML is).

**Comment by f-bader** [comment]
HI @gabriel-samfira sorry I missed the already closed incident on this.
Thanks for the prompt reply and action.

> A new release "protocol" will be created with pre-releases before a final release (if possible please help test out future pre-releases if possible).
Definitely will do. Will implement this pre-release in the testing environment.

>My apologies for breaking compatibility with no notice. I know this may have caused some hours of debugging. That is something I will try harder to avoid in the future.
Thank you for your continued development and support for this important module! It's the best YAML parser out there.

**Comment by gabriel-samfira** [comment]
A new version has been released: https://www.powershellgallery.com/packages/powershell-yaml/0.4.9

A new release has been generated on GitHub: https://github.com/cloudbase/powershell-yaml/releases/tag/v0.4.9

Please test it out and let me know if all is well.

**Comment by gabriel-samfira** [comment]
> Thank you for your continued development and support for this important module! It's the best YAML parser out there.

This humble module is a thin wrapper on top of the amazing [YamlDotNet](https://github.com/aaubry/YamlDotNet) library. They deserve all the appreciation and support they can get (especially considering they are a small team).

**Comment by gabriel-samfira** [comment]
Fixed in v0.4.9
